src/python/pants/engine/build_files.py

```diff
@@ -131,9 +131,9 @@ def collect_inline_dependencies(item):
   collect_inline_dependencies(struct)
 
   # And then hydrate the inline dependencies.
-  hydrated_inline_dependencies = yield [
+  hydrated_inline_dependencies: Tuple[HydratedStruct, ...] = await MultiGet(
```

Comment:
Hm... `await MultiGet` is unfortunate, but seems necessary. Might want to loop back to change the name.

Reply:
I spent an inordinately long time at the python repl to eventually get that `await` auto-flattens every coroutine chain, and that `yield`ing in `__await__` creates a generator. I didn't spend anywhere near as much time trying to make the syntax more appropriate. This can absolutely be revisited.

Reply:
I was DMing Danny that I think `GetMultiple` reads better. He explained that `MultiGet` was to mirror Rust code. Maybe there's another option too that would work better?

Reply:
This also maybe feels like an opportunity to go with something like`AllOf` (as opposed to `FirstOf`, etc)?

Should probably look at other python async frameworks for inspiration, but still best as a followup.